docs/contributor-guide/developer-guide/technical-contribution-guide.md

```diff
@@ -18,13 +18,86 @@ Development tasks are managed in the [GitHub issues](https://github.com/civiform
 
 If you're just getting started, check out issues labeled with [Good First Issue](https://github.com/civiform/civiform/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). Also check out issues in the [next milestone](https://github.com/civiform/civiform/milestones?direction=asc\&sort=due\_date\&state=open) so you can work on the highest-priority tasks.
 
-### Pull requests
+### Git workflow
 
-When you're ready to submit your code, open a pull request with "Closes #X" to link the relevant issue.
+To start working on a change, first get the latest state from the github repository then create a new local development branch:
+
+```sh
+# Downloads any commits from the github repository that your local repository does not have.
+git fetch origin
+
+# Creates a new local branch `your_username/dev-branch-description`.
+# The repository file contents are set to the same as the github repository's main branch.
+git checkout -b your_username/dev-branch-description origin/main
+```
+
+Then start making your changes. In general, smaller commits that make "one" logical change are easier to manage and change as development progresses. The 'Commit Guidelines' section in [this guide](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project) has more discussion on commit best practices. The repository is configured such that all commits in a pull request are squashed into one commit when the pull request is merged so having many commits poses no issues.
+
+When you make a change the change initially is 'unstaged'. To package a change into a commit you need to instruct git to 'stage' the change. See [this guide](https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository) for details on staged vs unstaged changes.
+
+```sh
+# Shows what branch you are on and what files you have modified. All changes will start as unstaged.
+git status
+
+# Stage changes for a commit.
+#
+# The add command supports many ways to specify what changes to stage. To stage only certain changes, use:
+git add -- path/to/file/1 path/to/file/2
+
+# To interactively choose what changes to stage, use:
+git add --patch
+
+# To stage all changes:
+git add .
+```
+
+Once some changes are staged, you can create a commit the packages them together with a description:
+
+```
+# Double-check only the desired changes are staged.
+git status
+
+# Create a new commit packaging the staged changes with the description `My change`.
+git commit --message 'My change'
```

Comment:
Going along with Kiana's comment, before doing the commit is when you'd want to diff changes. For unstaged `git diff` or for staged `git diff --staged`. Or `git difftool` if you have it configured to use an external application.